framework/auth/core.py

```diff
@@ -1376,13 +1380,22 @@ def is_affiliated_with_institution(self, inst):
         return inst in self.affiliated_institutions
 
     def remove_institution(self, inst_id):
-        try:
-            self.affiliated_institutions.remove(inst_id)
-        except ValueError:
-            return False
-        return True
+        node_id = False
+        for node in self._affiliated_institutions:
```

Comment:
Is there a reason you're iterating through the protected `_affiliated_institution` instead of `affiliated_institutions`?